accelerator/cpu_accelerator.py

```diff
@@ -0,0 +1,262 @@
+# Copyright (c) Microsoft Corporation.
+# SPDX-License-Identifier: Apache-2.0
+
+# DeepSpeed Team
+
+import torch
+from deepspeed.accelerator.abstract_accelerator import DeepSpeedAccelerator
+import oneccl_bindings_for_pytorch  # noqa: F401
+import psutil
+
+
+# accelerator for Intel CPU
+class CPU_Accelerator(DeepSpeedAccelerator):
+
+    def __init__(self):
+        self._name = 'cpu'
+        self._communication_backend_name = 'ccl'
+        self.max_mem = psutil.Process().memory_info().rss
+
+        def _check_and_mapping_mpich_env():
+            # Workaround when using Intel MPI launcher
+            # Model can check "RANK" and "WORLD_SIZE" for rank and world size
+            import os
+            if "RANK" not in os.environ and "PMI_RANK" in os.environ:
+                os.environ['RANK'] = os.environ.get('PMI_RANK')
+                print("mapping environment variable PMI_RANK to RANK")
+            if "WORLD_SIZE" not in os.environ and "PMI_SIZE" in os.environ:
+                os.environ['WORLD_SIZE'] = os.environ.get('PMI_SIZE')
+                print("mapping environment variable PMI_SIZE to WORLD_SIZE")
+            if "LOCAL_RANK" not in os.environ and "MPI_LOCALRANKID" in os.environ:
+                os.environ['LOCAL_RANK'] = os.environ.get('MPI_LOCALRANKID')
+                print("mapping environment variable MPI_LOCALRANKID to LOCAL_RANK")
+
+        _check_and_mapping_mpich_env()
+
+    def is_synchronized_device(self):
+        return True
+
+    # Device APIs
+    def device_name(self, device_index=None):
+        return 'cpu'
+
+    def device(self, device_index=None):
+        return None
+
+    def set_device(self, device_index):
+        return
+
+    def current_device(self):
+        return 0
+
+    def current_device_name(self):
+        return 'cpu'
+
+    def device_count(self):
```

Comment:
TODO: Number of devices on CPU could be flexible.  A more reasonable number is same number as number of sockets or sub numa clusters.   We should use numa util to help decide device_count for CPU Accelerator.

Reply:
fixed.